cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/FileLocalizer.java

```diff
@@ -86,13 +87,15 @@ public void run(String[] args) throws Exception {
           break;
         }
 
-        Location location = context.getLocationFactory().create(localFile.getURI());
+        Location localizedFile = new LocalLocationFactory().create(new File(localFile.getURI().getPath()).toURI());
+        fileFetcher.download(localFile.getURI(), localizedFile);
+
         Path targetPath = targetDir.resolve(localFile.getName());
 
         if (localFile.isArchive()) {
-          expand(location, targetPath);
+          expand(localizedFile, targetPath);
         } else {
-          copy(location, targetPath);
+          copy(localizedFile, targetPath);
```

Comment:
Shouldn't copy again. The file has been downloaded to local disk already. You should either link it, or better yet having the file fetcher write to this destination directly.